src/commontransformations.jl

```diff
@@ -0,0 +1,352 @@
+
+
+function transform(trans::Rotation, x::FixedVector{3})
+    (m, x2) = promote(trans.matrix, x)
+    (typeof(x2))(m * Vec(x2))
+end
+
+transform_deriv(trans::Rotation, x::FixedVector{3}) = trans.matrix # It's a linear transformation, so this is easy!
+
+function transform_deriv_params{T1,T2}(trans::Rotation{Void,T1}, x::FixedVector{3,T2})
+    # This derivative isn't projected into the orthogonal/Hermition tangent
+    # plane. It would be acheived by:
+    # Δ -> (Δ - R Δ' R) / 2
+
+    # The matrix gives 9 parameters...
+    Z = zero(promote_type(T1,T2))
+    @fsa [ x[1] x[2] x[3] Z Z Z Z Z Z;
+           Z Z Z x[1] x[2] x[3] Z Z Z;
+           Z Z Z Z Z Z x[1] x[2] x[3] ]
+end
+
+
+function transform_deriv_params{T1,T2}(trans::Rotation{Quaternion{T1},T1}, x::FixedVector{3,T2})
+    #=
+    # From Rotations.jl package
+    # get rotation matrix from quaternion
+    xx = q.v1 * q.v1
+    yy = q.v2 * q.v2
+    zz = q.v3 * q.v3
+    xy = q.v1 * q.v2
+    zw = q.s  * q.v3
+    xz = q.v1 * q.v3
+    yw = q.v2 * q.s
+    yz = q.v2 * q.v3
+    xw = q.s  * q.v1
+
+    # initialize rotation part
+    return @fsa([1 - 2 * (yy + zz)    2 * (xy - zw)       2 * (xz + yw);
+                 2 * (xy + zw)        1 - 2 * (xx + zz)   2 * (yz - xw);
+                 2 * (xz - yw)        2 * (yz + xw)       1 - 2 * (xx + yy)])
+
+    # This leads to these derivative matrices:
+
+    dR/ds =
+    [ 0    -2*v3    2*v2;
+     2*v3     0    -2*v1;
+    -2*v2  2*v1     0   ]
+
+    dR/dv1 =
+    [ 0  2*v2  2*v3 ;
+     2*v2 -4*v1 -2*s;
+     2*v3  2*s  -4*v1]
+
+    dR/dv2 =
+    [-4*v2  2*v1  2*s;
+      2*v1  0     2*v3;
+     -2*s   2*v3  -4*v2]
+
+    dR/dv3 =
+    [-4*v3  -2*s  2*v1;
+     2*s    -4*v3  2*v2;
+     2*v1   2*v2   0 ]
+    =#
+    s = trans.rotation.s
+    v1 = trans.rotation.v1
+    v2 = trans.rotation.v2
+    v3 = trans.rotation.v3
+
+    @fsa [ 2(-v3*x[2]+v2*x[3])  2(v2*x[2]+v3*x[3])            2(-2*v2*x[1]+v1*x[2]+s*x[3])  2(-2*v3*x[1]-s*x[2]+v1*x[3]) ;
+           2(v3*x[1]-v1*x[3])   2(v2*x[1]-2*v1*x[2]- s*x[3])  2(v1*x[1]+v3*x[3])            2(s*x[1]-2*v3*x[2]+v2*x[3])  ;
+           2(-v2*x[1]+v1*x[2])  2(v3*x[1]+ s*x[2]-2*v1*x[3])  2(-s*x[1]+v3*x[2]-2*v2*x[3])  2(v1*x[1]+v2*x[2])           ]
+end
+
+#function transform_deriv_params{T1,T2}(trans::Rotation{EulerAngles{T1},T1}, x::FixedVector{3,T2})
+#end
+
+Base.inv(trans::Rotation) = Rotation(nothing, trans.matrix')
+Base.inv{T <: Union{Quaternion, SpQuat}}(trans::Rotation{T}) = Rotation(inv(trans.rotation), trans.matrix')
+
+compose(t1::Rotation, t2::Rotation) = Rotation(nothing, t1.matrix*t2.matrix) # A bit lossy for transform_deriv_params, but otherwise probably the most efficient choice
+
+
+#############################
+# Individual axis rotations #
+#############################
+immutable RotationXY{T} <: AbstractTransformation{FixedVector{3}, FixedVector{3}}
+    angle::T
+    sin::T
+    cos::T
+end
```

Comment:
Hmm.  `Rotation{i,j}` would probably be cool, as long as we had some easy to use type aliases for rotations in 2D and 3D.

Oh, you could also construct left hand rotations this way - `Rotation{2,1}` anyone?

Unsure how to make the type checking nice, but keep it all lightweight.  `Rotation{i,j,Dim}` is the obvious possibility and with a couple of type aliases it would be pretty usable for the usual things I guess -

```julia
typealias Rotation2D Rotation{1,2,2}
typealias RotationZX Rotation{3,1,3}
# etc
```
?